### ISSUE TITLE
Missing implication operator in `Data.Bool.Base`

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1632]
Implication, like `xor` is derivable. We go through various hoops to derive `xor` generically for any boolean algebra below:

https://github.com/agda/agda-stdlib/blob/69c808fa1491eaff678cb9e29961e6068ef9f1ad/src/Algebra/Lattice/Properties/BooleanAlgebra.agda#L246-L249

I wonder if there's a middle ground.

**Comment by JacquesCarette** [comment]
Can you expand on what you mean by "a middle ground" here?

**Comment by MatthewDaggitt** [comment]
No, I can't. I no longer remember, apologies. I think we should just add it to `Data.Bool` and then prove properties directly. We can still keep the existing code above.
